test/e2e/cluster.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# shellcheck source=/dev/null
+source "$(dirname "$0")/utils.sh"
+
+cd_root_path
+
+# Build
+export GO111MODULE=on
+go build
+
+# Prepare container
+trap "docker rm -f etcdadm-{0,1,2};rm -f ${PWD}/bin/ca.*" EXIT
+
+for ((i=0;i<3;i++))
+do
+    docker run --name etcdadm-${i} --detach --privileged --security-opt seccomp=unconfined --tmpfs /tmp --tmpfs /run --volume ${PWD}:/etcdadm kindest/node:v1.16.2
+done
+
+# Run init
+docker exec etcdadm-0 /etcdadm/etcdadm init
+
+# Verify that all endpoints are healthy
+docker exec etcdadm-0 /opt/bin/etcdctl.sh endpoint health
+
+# Verify that init container ip address
+etcdadm_0_ip=$(docker inspect --format {{.NetworkSettings.Networks.bridge.IPAddress}} etcdadm-0)
+
+# Copy CA certs from etcdadm-0 container to bin directory
+docker cp etcdadm-0:/etc/etcd/pki/ca.crt ${PWD}/bin/
+docker cp etcdadm-0:/etc/etcd/pki/ca.key ${PWD}/bin/
+
+# Add other member
+for ((i=1;i<3;i++))
+do
+    echo "Copy CA certs to container etcdadm-${i}"
+    # Copy CA certs to container
+    docker exec etcdadm-${i} mkdir -p /etc/etcd/pki
+    docker cp ${PWD}/bin/ca.crt etcdadm-${i}:/etc/etcd/pki/
+    docker cp ${PWD}/bin/ca.key etcdadm-${i}:/etc/etcd/pki/
+
+    echo "Join etcd member etcdadm-${i}"
+    docker exec etcdadm-${i} /etcdadm/etcdadm join https://${etcdadm_0_ip}:2379 --name etcdadm-${i}
+    docker exec etcdadm-${i} /opt/bin/etcdctl.sh endpoint health --cluster -w table
+
+    docker exec etcdadm-0 /opt/bin/etcdctl.sh endpoint health --cluster -w table
```

Comment:
why do we need to call the etcdctl on the `init` container every time a member joined?
i think this can be omitted.

Reply:
Yes, you are right, we don't need to call the etcdctl on `init` container every time, I only want to print etcd cluster endpoints in `init` container.